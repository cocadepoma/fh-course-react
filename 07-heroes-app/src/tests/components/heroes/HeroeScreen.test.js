import React from "react";
import { MemoryRouter, Route } from "react-router-dom";
const { mount } = require("enzyme");
const { HeroeScreen } = require("../../../components/heroes/HeroeScreen");
// Para poder simular los params por URL hay que envolver el componente en una
// Route fictia, en la que le indicamos el path, y el componente que debe renderizar
// Este Route debe de ir envuelto en un MemoryRouter inicializado con initialEntries
// indicando el path con los params.
describe("pruebas en <HeroeScreen/>", () => {
    const historyMock = {
        length: 10,
        push: jest.fn(),
        goBack: jest.fn(),
    };

    const wrapper = mount(
        <MemoryRouter initialEntries={["/hero"]}>
            <HeroeScreen history={historyMock} />
        </MemoryRouter>
    );

    test("debe de mostrar el componente redirect si no hay argumentos en el URL", () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("Redirect").exists()).toBe(true);
    });

    test("debe de regresar a la pantalla anterior con PUSH", () => {
        const historyMock = {
            length: 1,
            push: jest.fn(),
            goBack: jest.fn(),
        };
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
                <Route
                    path='/hero/:heroeId'
                    component={() => <HeroeScreen history={historyMock} />}
                />
            </MemoryRouter>
        );

        wrapper.find("button").prop("onClick")();

        expect(historyMock.push).toBeCalledTimes(1);
        expect(historyMock.push).toBeCalledWith("/");
        expect(historyMock.goBack).not.toBeCalled();
    });

    test("debe de redireccionar a la pantalla anterior con GOBACK", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
                <Route
                    path='/hero/:heroeId'
                    component={() => <HeroeScreen history={historyMock} />}
                />
            </MemoryRouter>
        );

        wrapper.find("button").prop("onClick")();

        expect(historyMock.goBack).toBeCalled();
        expect(historyMock.push).not.toBeCalled();
    });
    test("debe de mostrar un heroe si el parÃ¡metro existe", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
                <Route path='/hero/:heroeId' component={HeroeScreen} />
            </MemoryRouter>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(".row").exists()).toBe(true);
    });
    test("Debe de llamar el redirect si el heroe no existe", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero/marvel-spiderssss"]}>
                <Route path='/hero/:heroeId' component={HeroeScreen} />
            </MemoryRouter>
        );

        expect(wrapper.text()).toBe("");
    });
});
