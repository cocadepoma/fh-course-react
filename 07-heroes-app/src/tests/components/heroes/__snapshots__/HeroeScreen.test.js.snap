// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HeroeScreen/> debe de mostrar el componente redirect si no hay argumentos en el URL 1`] = `""`;

exports[`pruebas en <HeroeScreen/> debe de mostrar un heroe si el par√°metro existe 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4 animate__animated animate__fadeInLeft animate__faster"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail"
      src="../assets/heroes/marvel-spider.jpg"
    />
  </div>
  <div
    className="col-8"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <b>
          Alter ego:
        </b>
         
        Peter Parker
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher:
        </b>
         
        Marvel Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First Appearance:
        </b>
         
        Amazing Fantasy #15
      </li>
    </ul>
    <h5>
      Characters:
    </h5>
    <p>
      Peter Parker
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
